### PR TITLE
Add Camel Mock

### PR DESCRIPTION
**Pull request by thelooter**
# Description
Adds a Mock for Camels

# Checklist
The following items should be checked before the pull request can be merged.
- [x] Code follows existing style.
- [x] Unit tests added (if applicable).

### CODE REVIEW

File: src/main/java/be/seeseemelk/mockbukkit/entity/CamelMock.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+import java.util.UUID;
+
+public class CamelMock extends AbstractHorseMock implements Camel
+{
+
+	private boolean isDashing = false;
+	private boolean isSitting = false;
+
+	/**
+	 * Constructs a new {@link AbstractHorseMock} on the provided {@link ServerMock} with a specified {@link UUID}.
```
Comment (Insprill) [inline src/main/java/be/seeseemelk/mockbukkit/entity/CamelMock.java]: Please update this Javadoc with the correct class name

**Review by Insprill — CHANGES_REQUESTED** [review]

File: src/test/java/be/seeseemelk/mockbukkit/entity/CamelMockTest.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+	{
+		assertFalse(camel.isDashing());
+	}
+
+	@Test
+	void testIsDashing()
+	{
+		camel.setDashing(true);
+		assertTrue(camel.isDashing());
+	}
```
Comment (tatanpoker09) [inline src/test/java/be/seeseemelk/mockbukkit/entity/CamelMockTest.java:56]: We could just theoretically include a test to see if setting dashing back to false also reverts back the isDashing condition, might be a little overkill and unnecessary, it does cover the other condition.

File: src/test/java/be/seeseemelk/mockbukkit/entity/CamelMockTest.java [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+	{
+		assertFalse(camel.isSitting());
+	}
+
+	@Test
+	void testSetSitting()
+	{
+		camel.setSitting(true);
+		assertTrue(camel.isSitting());
+	}
```
Comment (tatanpoker09) [inline src/test/java/be/seeseemelk/mockbukkit/entity/CamelMockTest.java:69]: Same here with sitting, up to your decision. Other than that LGTM

**Review by tatanpoker09 — APPROVED** [review]
LGTM. Couple minor things that *might* be considerable, but it looks good.

File: src/test/java/be/seeseemelk/mockbukkit/entity/CamelMockTest.java (hunk 2 above)
Comment (Insprill) [reply src/test/java/be/seeseemelk/mockbukkit/entity/CamelMockTest.java:56]: We don't do that anywhere else, so it'd have to be updated everywhere for consistency, and that's out of the scope of this PR. The odds of that being an issue are extremely negligible, so we'll likely never do it.

**Review by Insprill — APPROVED** [review]
LGTM

**Pull request merged**